src/zenml/alerter/base_alerter.py

```diff
@@ -0,0 +1,38 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import ClassVar
+
+from zenml.enums import StackComponentType
+from zenml.stack import StackComponent
+
+
+class BaseAlerter(StackComponent, ABC):
+    """Base class for all ZenML alerters."""
+
+    # Class configuration
+    TYPE: ClassVar[StackComponentType] = StackComponentType.ALERTER
+    FLAVOR: ClassVar[str]
+
+    @abstractmethod
+    def post(self, message: str) -> bool:
```

Comment:
An idea for a BaseMethod:
```
def ask(question: str, metadata: Dict[str, str]) -> bool:
    """Asks a question to alerter and waits for a boolean response. Metadata is a dict of key value pairs that are optionally  published to the channel"""

```

Reply:
Resolving this, you already implemented it :)